subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBaseApiService.java

```diff
@@ -669,6 +670,51 @@ public int handleBasePlanEvent(final DefaultSubscriptionBase subscription, final
         }
     }
 
+    @Override
+    public int handleExpiredEvent(final DefaultSubscriptionBase subscription, final SubscriptionBaseEvent event, final SubscriptionCatalog catalog, final CallContext context) throws CatalogApiException {
+
+        final InternalCallContext internalCallContext = createCallContextFromBundleId(subscription.getBundleId(), context);
+        if (event.getType() == EventType.EXPIRED && subscription.getCategory() == ProductCategory.BASE) {
```

Comment:
Do we need the check `event.getType() == EventType.EXPIRED`, isn't it guaranteed `handleExpiredEvent` is only invoked for `EXPIRED` notifications? Also, if we were to keep it, it would then be missing in the `else if ` down.

Reply:
Agreed, I've removed this check.